pkg/tfgen/generate_go.go

```diff
@@ -253,11 +266,41 @@ func (g *goGenerator) emitConfigAccessor(w *tools.GenWriter, v *variable) {
 	} else if v.rawdoc != "" {
 		g.emitRawDocComment(w, v.rawdoc, "")
 	}
+
+	defaultValue := ""
+	if v.optional() {
+		defaultValue = g.goDefaultValue(v)
+	}
+
 	w.Writefmtln("func Get%s(ctx *pulumi.Context) %s {", upperFirst(v.name), gettype)
-	w.Writefmtln("\treturn config.%s(ctx, \"%s:%s\")", getfunc, g.pkg, v.name)
+
+	configKey := fmt.Sprintf("\"%s:%s\"", g.pkg, v.name)
+	if defaultValue != "" {
+		w.Writefmtln("\tif v := config.Get(ctx, %s); v != \"\" {", configKey)
+		w.Writefmtln("\t\treturn config.%s(ctx, %s)", getfunc, configKey)
```

Comment:
In the generated AWS diff, it was surprising that this leads to duplicated `config.Get(ctx, "aws:region")`.  Not a big deal though.